File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = ["a","e","i","o","u"]
        l,r = 0,len(s)-1
        new_s = list(s)
    
        while (l<r):
            v_l = new_s[l].lower() in vowels
            v_r = new_s[r].lower() in vowels
            if v_l and v_r:
                new_s[l],new_s[r] = new_s[r],new_s[l]
                l+=1
                r-=1
            elif v_l:
                r-=1
            elif v_r:
                l+=1
            else:
                l+=1
                r-=1
            
        return ("").join(new_s)

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.ts
function reverseVowels(s: string): string {
  let curr = s.split("")
 
    let l = 0
    let r = s.length-1
    const letters = ["a","e","i","o","u"]
    const vowels =new  Set(letters);
    
    while(l<=r){
        if (vowels.has(s[l].toLowerCase())&& vowels.has(s[r].toLowerCase())){
            curr[l] = s[r]
            curr[r] = s[l]
            l++;
            r--;
        }
        else if (vowels.has(s[l].toLowerCase())){
            r--;
        }
        else if (vowels.has(s[r].toLowerCase())){
            l++;
        }
        else{
            l++;
            r--;
        }
    }


  return curr.join("")  
};